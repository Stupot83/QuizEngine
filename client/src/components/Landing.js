import React from "react";
import "../sass/Landing.scss";
import { Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Logo from "../Logo.png";
import Button from "@material-ui/core/Button";
import CreateIcon from "@material-ui/icons/Create";
import LockOpenIcon from "@material-ui/icons/LockOpen";

const useStyles = makeStyles(theme => ({
    root: {
        flexGrow: 1
    },
    headerGrid: {
        minHeight: "10vh",
        maxHeight: "10vh",
        textAlign: "center"
    },
    logoGrid: {
        minHeight: "50vh",
        maxHeight: "50vh"
    },
    buttonGrid: {
        minHeight: "30vh",
        maxHeight: "30vh"
    }
}));

export default function Landing() {
    const classes = useStyles();

    return (
        <div className={(classes.root, "Page_container")}>
            <Container className="Landing_container">
                <Grid container spacing={3}>
                    <Grid className={classes.headerGrid} item xs={12}>
                        <h1 id="Landing_header" className="Landing_header">
                            Welcome to QuizEngine
                        </h1>
                    </Grid>
                    <Grid className={classes.logoGrid} item xs={12}>
                        <img id="Landing_logo" className="Landing_logo" src={Logo} alt="Logo" />
                    </Grid>
                    <Grid container spacing={3} className="Landing_button_container">
                        <Grid className={classes.buttonGrid} item xs={12} className="Landing_button_box">
                            <Button
                                component={Link}
                                id="Landing_registration_button"
                                className="Landing_button_register"
                                variant="contained"
                                size="large"
                                startIcon={<CreateIcon />}
                                to="/register"
                            >
                                Register
                            </Button>
                            <Button
                                component={Link}
                                id="Landing_login_button"
                                className="Landing_button_login"
                                variant="contained"
                                size="large"
                                startIcon={<LockOpenIcon />}
                                to="/login"
                            >
                                Login
                            </Button>
                        </Grid>
                    </Grid>
                </Grid>
            </Container>
        </div>
    );
}
