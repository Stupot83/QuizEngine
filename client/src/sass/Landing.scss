.root {
 background-color: ghostwhite;
 background-image: url("https://www.transparenttextures.com/patterns/brick-wall.png");
 min-height: 100vh;
}

.Landing_container {
 padding-top: 64px;

 .Landing_header {
  font-weight: 900 !important;
 }

 .Landing_logo {
     margin-top: 15px;
     height: 100%;
     display: block;
     margin-left: auto;
     margin-right: auto;
     border-radius: 60px;
 }

 .Landing_button_container {
  
  
  .Landing_button_box {
   text-align: center;
   padding-top: 70px;
   padding-bottom: 100px;

   .Landing_button_register {
    min-width: 150px;
    margin-left: 30px;
    margin-right: 30px;
    padding: 20px;
    border-radius: 15px;
    background-color: #D1345B;
    color: ghostwhite;
    font-weight: bold;
}

.Landing_button_login {
 min-width: 150px;
 margin-left: 30px;
 margin-right: 30px;
 padding: 20px;
 border-radius: 15px;
 background-color: #3454D1;
 color: ghostwhite;
 font-weight: bold;
}
 }
 }
}