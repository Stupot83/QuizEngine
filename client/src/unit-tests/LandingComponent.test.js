import React from "react";
import { shallow } from "enzyme";
import Landing from "../components/Landing";

describe("Landing component", () => {
    it("should render correctly in debug mode", () => {
        const component = shallow(<Landing debug />);

        expect(component).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const component = shallow(<Landing />);

        expect(component).toMatchSnapshot();
    });
});
