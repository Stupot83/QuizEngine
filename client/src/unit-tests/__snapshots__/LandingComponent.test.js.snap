// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component should render correctly 1`] = `ShallowWrapper {}`;

exports[`Landing component should render correctly in debug mode 1`] = `ShallowWrapper {}`;
